### ISSUE TITLE
Configurable, Science Aware Tool for Labelling Clouds

### ISSUE THREAD

**Issue by BenGalewsky** [ESIPFed/GSoC#27]
## ESIP Member Organization
National Center for Supercomputing Applications (NCSA)

## Mentor
Ben Galewsky

## Project Idea
Engineer a jupyter notebook that allows researchers to build custom tools to label clouds from Earth Observing Systems such as the Terra Satellite. 

## Information for students
Just the generally information for students. See [ESIP Student Guide](https://github.com/ESIPFed/gsoc/blob/master/STUDENT-contribution-guide.md)

## Abstract
Clouds are the greatest source of climate change uncertainty. The twenty years of Terra satellite data gives us an incredible opportunity to train sophisticated models to further understand climate processes. In order to develop these models, we need an extensive set of labelled data based on the various instruments housed on the Terra and Aqua satellites. Labelling clouds can be difficult, especially against and snowy backdrop. 

This task can be improved by considering images captured in different wavelengths, or from spatial shifts. Further improvements can be gained by considering the physics of the underlying elements. 

We believe that the best way to create labelling tools is to provide a highly customizable platform that can be programmed by researchers using the familiar python and Jupyter Notebook environments. We are working with the [geoviews](https://geoviews.org) python library to build just such a framework. 

There is plenty of data available from Terra and Aqua satellites and this approach could be generalized to many other areas of science.

The general idea is to think through the user experience to give researchers powerful tools to label the clouds. There are multiple images available for a given location on the earth. Users should be able to flip through, overlay, or display them side-by-side. You might want to implement a "Magic Wand" style selection to create a polygon on one layer. Then users could adjust the polygon given pixels found on other layers. Assume that scientists will want to create numeric tools to give more hints based on their knowledge of the physical processes.

## Technical stuff
Python, Jupyter Notebooks, Conda

Helpful Experience, but not required!
Geospatial Information Systems (GIS), Javascript

## First steps
Try out [our demo](https://mybinder.org/v2/gh/BenGalewsky/ml-labeling/master?filepath=ml-annotators.ipynb) from the ESIP Winter Meeting on BinderHub

Watch [this talk](https://slideslive.com/38922389/cumulo-a-dataset-for-learning-cloud-classes) on the importance and challenges of labelling cloud data

*Example Images*
Here are [some examples](https://uofi.box.com/s/ayf9ejz9nirlg9878q8u54mv43rno5b7) of images of the same area captured with different instruments or wavelengths.

Learn a bit more about the instruments:
* [ASTER](https://terra.nasa.gov/about/terra-instruments/aster)
* [CERES](https://terra.nasa.gov/about/terra-instruments/ceres)
* [MISR](https://terra.nasa.gov/about/terra-instruments/misr)
* [MODIS](https://terra.nasa.gov/about/terra-instruments/modis)
* [MOPITT](https://terra.nasa.gov/about/terra-instruments/mopitt)

**Comment by snack0verflow** [comment]
Hi @BenGalewsky just wanted to know if there are any evaluative tasks that I need to do, apart from the ones listed here? Asking in context of GSoC 2020.
I have some experience in Deep Learning and would love to work on this project. :))

**Comment by snack0verflow** [comment]
Also could you please suggest some active repositories in https://github.com/ESIPFed on which I can start contributing. Thanks :)

**Comment by 0xamogh** [comment]
@BenGalewsky 
Hey! The work going on this organisation seems really interesting. I had a question regarding the above idea. Does it involve exploring and training data or is it more on the lines of building tools which could facilitate others to do ML

**Comment by BenGalewsky** [comment]
The focus of this project is to develop powerful labelling tools to create the training sets needed by climate science. If you want to use them for developing your own models feel free, but that is not part of this project.

**Comment by punitgalav** [comment]
Hi @BenGalewsky , I am Punit Galav currently pursuing Masters in Geoinformatics and Natural Resources Engineering of IIT Bombay. I have all the relevant skills required for this project and have studied GIS, Remote sensing of natural resources, Scattering models, Machine learning. I understand the importance of developing such a framework and looking forward to contribute to this project.
